### DISCUSSION TITLE
How to force outgoing mails to use a fixed From: header?

### DISCUSSION THREAD

**Discussion by franco-giordano** [simple-login/app#1062] [Self-hosting question]
Hi! After a few days I managed to self-host SimpleLogin in Oracle Cloud. The problem is that [OCI blocks outbound port 25](https://docs.oracle.com/en-us/iaas/releasenotes/changes/f7e95770-9844-43db-916c-6ccbaf2cfe24/), so I must use their Email Service to send outgoing mails.

All good right until I realized that I can't just send with any mail in my domain, but that [I must pre-approve each outbound email address I use](https://docs.oracle.com/en-us/iaas/Content/Email/Tasks/managingapprovedsenders.htm). The problem is that SimpleLogin uses different addresses for everything (relaying, bounces, transactions, etc).

Is there any way to use a fixed From: header in outbound emails? I've been trying to use `smtp_header_checks` in `main.cf` like this:
```
# /etc/postfix/main.cf
smtp_header_checks = regexp:/etc/postfix/header_checks
```
```
# /etc/postfix/header_checks
/^From:[[:space:]]+(.*)[[:space:]]<(.*)@mydomain.com>/ REPLACE From: ${1} <outgoing@mydomain.com>
```

The idea is that I would only need to use `outgoing@mydomain.com` as an Approved Sender, so no emails are rejected by OCI, but no luck. Any ideas?

**Comment by franco-giordano** [comment]
Hey @nguyenkims, sorry for the ping, but any ideas on this? I'm really struggling on this one. Is there any setting on SimpleLogin to force this? Or maybe just fork the project and change the code, if it's simple enough

**Reply by nguyenkims** [reply]
@franco-giordano SimpleLogin doesn't have any way to force a fixed From header. Emails need to be sent from a different "reverse alias" so you can reply to a forwarded email and reverse alias is different for each sender.

**Reply by franco-giordano** [reply]
I see, no luck then. Thanks for your help!
